cloudify_types/cloudify_types/component/operations.py

```diff
@@ -66,16 +66,52 @@ def _get_desired_operation_input(key, args):
 def _get_client(kwargs):
     client_config = _get_desired_operation_input('client', kwargs)
     if client_config:
-        return CloudifyClient(**client_config)
+        client = CloudifyClient(**client_config)
+
+        # for determining an external client:
+        manager_ips = [mgr.private_ip for mgr in
+                       manager.get_rest_client().manager.get_managers()]
+        internal_hosts = ({'127.0.0.1', 'localhost'} | set(manager_ips))
+        host = {client.host} if type(client.host) == str \
+            else set(client.host)
+        is_external_host = not (host & internal_hosts)
```

Comment:
you know what would be funny, if `host` was several entries and only _some_ matched our own manager, and others didn't. What would that mean? Who knows. Maybe we want to at least log a warning.